modules/standard/List.chpl

```diff
@@ -1318,20 +1324,152 @@ module List {
       return;
     }
 
+    /*
+      Return a copy of the element at a given index in this list.
+
+      :arg i: The index of the element to get.
+
+      .. warning::
+
+        Use of the `getValue` method with an out of bounds index (while
+        bounds checking is on) will cause the currently running program
+        to halt.
+
+      :return: A copy of an element from this list.
+    */
+    proc const getValue(i: int): eltType {
+      _enter(); defer _leave();
+
+      if boundsChecking && !_withinBounds(i) {
+        const msg = "Invalid list index: " + i:string;
+        boundsCheckHalt(msg);
+      }
+
+      return _getRef(i);
+    }
+
+    /*
+      Return a borrow of the element at a given index in this list. This
+      method can only be called when this list's element type is a class
+      type.
+
+      :arg i: The index of the element to borrow.
+      :type i: `int`
+
+      :return: A borrow of an element from this list.
+    */
+    proc const getBorrowed(i: int) where isClass(eltType) {
+      _enter(); defer _leave();
+
+      if boundsChecking && !_withinBounds(i) {
+        const msg = "Invalid list index: " + i:string;
+        boundsCheckHalt(msg);
+      }
+
+      ref slot = _getRef(i);
+
+      return slot.borrow();
+    }
+
+    /*
+      Sets the element at a given index in this list. This method returns
+      `false` if the index is out of bounds.
+
+      :arg i: The index of the element to set
+      :type i: int
+
+      :arg x: The value to set at index `i`
+
+      :return: `true` if `i` is a valid index that has been set to `x`,
+               and `false` otherwise.
+      :rtype: bool
+    */
+    proc ref set(i: int, pragma "no auto destroy" in x: eltType): bool {
+      _enter(); defer _leave();
+
+      if !_withinBounds(i) {
+        _destroy(x);
+        return false;
+      }
+
+      ref src = x;
+      ref dst = _getRef(i);
+      _move(src, dst);
```

Comment:
You have to call `_destroy(dst)` before the move

Reply:
Thanks!